src/parsers/InputFieldDeclarationParser.ts

```diff
@@ -56,7 +81,60 @@ export class InputFieldDeclarationParser {
 
 	static templates: Template[] = [];
 
-	static parse(fullDeclaration: string): InputFieldDeclaration {
+	static parseDeclaration(
+		input: InputFieldDeclaration,
+		args: Record<InputFieldArgumentType, string> | {} | undefined | null = undefined,
+		templateName: string | undefined | null = undefined
+	): InputFieldDeclaration {
+		// field type check
+		input.inputFieldType = InputFieldDeclarationParser.getInputFieldType(input.inputFieldType);
+
+		// template check:
+		let useTemplate = isTruthy(templateName) && typeof templateName === "string";
+		if (useTemplate) {
```

Comment:
since this is the same as for `parseString()` it is with thinking about extracting this into it's separate method

Reply:
I moved the logic to applyTemplate